aggregator/src/aggregator/accumulator.rs

```diff
@@ -121,6 +121,52 @@ impl<const SEED_SIZE: usize, Q: AccumulableQueryType, A: vdaf::Aggregator<SEED_S
         rslt
     }
 
+    /// Returns the set of report IDs that cannot be merged into its batch aggregation because the
+    /// batch has already been collected.
+    pub async fn unwritable_reports<C: Clock>(
+        &self,
+        tx: &Transaction<'_, C>,
+        vdaf: &A,
+    ) -> Result<HashSet<ReportId>, datastore::Error> {
+        let unmergeable_report_ids = Arc::new(Mutex::new(HashSet::new()));
+        try_join_all(self.aggregations.values().map(|data| {
+            let unmergeable_report_ids = Arc::clone(&unmergeable_report_ids);
+            async move {
+                tx.get_batch_aggregation::<SEED_SIZE, Q, A>(
+                    vdaf,
+                    data.batch_aggregation.task_id(),
+                    data.batch_aggregation.batch_identifier(),
+                    data.batch_aggregation.aggregation_parameter(),
+                    data.batch_aggregation.ord(),
+                )
+                .await
+                .map(|batch_aggregation| {
+                    batch_aggregation.map(|batch_aggregation| {
+                        if batch_aggregation.state() == &BatchAggregationState::Collected {
+                            // Unwrap safety: this only panics if the mutex is poisoned. If
+                            // it is, one of the other futures also panicked, so we can
+                            // panic too.
+                            unmergeable_report_ids
+                                .lock()
+                                .unwrap()
```

Comment:
Do we generally want to comment on mutex poisoning? My impression is that `.unwrap()` on a mutex to panic on mutex poisoning is standard in Rust code.

Reply:
I agree that panicking here is OK. I do still like to have an unwrap safety comment, but we could make each such comment smaller by adding a note to the code style section of `docs/DEVELOPMENT.md`, so that then you wouldn't have to explain it again each time.